### PR TITLE
return "No such device" instead of unknown error for the invalid GPU device number

### PR DESCRIPTION
**Pull request by JamesWanglf**
When I try to test the GPU transcoding from [lpms#280](https://github.com/livepeer/lpms/pull/280), the test fails. If an invalid GPU device number (eg 9999) is entered, it will fail with an unknown error from ffmpeg. Returning a "No such device" error code instead of an unknown error should solve the problem.

### CODE REVIEW

File: libavutil/hwcontext_cuda.c
Comment (cyberj0g) [inline libavutil/hwcontext_cuda.c]: Whether error is retriable or not makes sense only on Broadcaster level when deciding if something is wrong with O+T, or the segment, so it shouldn't be mentioned here in a separate project. We might want to create an upstream PR for these changes later.

File: libavutil/hwcontext_cuda.c
Comment (cyberj0g) [inline libavutil/hwcontext_cuda.c]: I suggested `ENXIO` as less widely-used code with a similar caption to reduce the chance of ambigious error messages. But good call on using `ENODEV`, it's probably fine.

**Review by cyberj0g — CHANGES_REQUESTED** [review]
Overall LGTM, please see the comments.

**Review by oscar-davids — APPROVED** [review]
LGTM!

**Review by cyberj0g — APPROVED** [review]

**Comment by cyberj0g** [comment]
LGTM! After merging, please create a separate PR to go-livepeer with updated commit id of Ffmpeg in install_ffmpeg.sh.

**Pull request merged**
